src/main/java/com/e_gineering/maven/gitflowhelper/EnforceVersionsMojo.java

```diff
@@ -1,49 +1,145 @@
 package com.e_gineering.maven.gitflowhelper;
 
+import org.apache.maven.artifact.Artifact;
 import org.apache.maven.artifact.ArtifactUtils;
 import org.apache.maven.plugin.MojoExecutionException;
 import org.apache.maven.plugin.MojoFailureException;
 import org.apache.maven.plugins.annotations.LifecyclePhase;
 import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.plugins.annotations.ResolutionScope;
+import org.apache.maven.project.MavenProject;
 
+import java.util.HashSet;
+import java.util.Set;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
 /**
  * If there is an ${env.GIT_BRANCH} property, assert that the current ${project.version} is semantically correct for the
- * git branch.
+ * git branch. Also, make sure there are no SNAPSHOT (plugin) dependencies.
  */
-@Mojo(name = "enforce-versions", defaultPhase = LifecyclePhase.VALIDATE)
+@Mojo(requiresDependencyCollection = ResolutionScope.TEST, name = "enforce-versions", defaultPhase = LifecyclePhase.VALIDATE)
 public class EnforceVersionsMojo extends AbstractGitflowBranchMojo {
 
+    @Parameter(defaultValue = "true", property = "enforceNonSnapshots", required = true)
```

Comment:
Documentation needed for this option.  I had to look at the code to find the name of the property to set when I wanted to turn this off.